nixos/modules/services/web-apps/immich.md

```diff
@@ -0,0 +1,277 @@
+
+*Upstream documentation:* <https://immich.app/docs/overview/introduction/>
+
+[Immich](https://immich.app/) is a self-hosted photo and video management solution.
+
+## Configuring {#module-immich-configuring}
+
+A complete list of options for the Immich module may be found
+[here](#opt-services.immich.enable).
+
+### Basic setup {#module-immich-configuring-basic-setup}
+
+Using the following configuration you can setup a basic installation of Immich:
+
+```nix
+{
+  services.immich = {
+    enable = true;
+
+    # Optionally set a different location for your media files
+    mediaLocation = "/my/immich/media";
+  };
+}
+```
+
+This will configure Immich and its components as well as Postgres and
+Redis databases and users for Immich.
+
+## Migrating from Upstream Docker Compose configuration {#module-immich-migrating-from-docker-compose}
+
+If you have been running Immich using the Docker Compose setup as
+recommended by upstream, you can follow this guide to migrate your
+existing installation to NixOS.
+
+This guide is going to reference variables in its examples, so make
+sure to either set those in your shell or replace them with the actual
+values.
+
+### Assumptions {#module-immich-migrating-from-docker-compose-assumptions}
+
+These are assumptions and goals of this migration guide:
+
+1. Your current installation's database name and username is `immich`.
+2. Your NixOS installation's database name and username are going to
+   remain the same.
+3. Your current library is located at `UPLOAD_LOCATION` as defined in
+   your current `.env` file.
+4. Your NixOS installation's library will be located at
+   `TARGET_LOCATION`, which may be the same as `UPLOAD_LOCATION`.
+5. Both `UPLOAD_LOCATION` and `TARGET_LOCATION` are absolute paths.
+
+### Stop current Immich installation {#module-immich-migrating-from-docker-compose-stop-immich}
+
+To ensure data consistency, you should stop your current Immich
+instance. Otherwise, your backups/database dumps might not be in
+sync. Make sure to keep your current Postgres instance running, so we
+can create database dumps later.
+
+Example:
+```console
+# docker stop immich_server immich_machine_learning
+```
+
+### Create a backup {#module-immich-migrating-from-docker-compose-backup}
+
+First you want to make a backup of your existing database and photo
+library. To do this, make a copy of or use a backup tool on
+`UPLOAD_LOCATION`.
+
+Example:
+
+```console
+# tar --create --zstd --file /var/backup/immich-library.tar.zst $UPLOAD_LOCATION
+```
+
+Next you should create a backup of your existing database. While this
+guide is not going to modify the original database it might still be
+advisable in case something goes wrong.
+
+Example, assuming your Immich database is running in a container
+named `immich_postgres`:
+
+```console
+# docker exec immich_postgres pg_dump --username immich immich -f /tmp/immich.sql
+# docker cp immich_postgres:/tmp/immich.sql /var/backup/immich.sql
```

Comment:
```suggestion
# docker exec immich_postgres pg_dump --username immich immich > /tmp/immich.sql
```

Reply:
I had bad experiences with piping content to/from Docker commands. If you ever add `-t` to a docker exec call, all output will be combined into `stdout` so your piped data would include errors/warnings too.
Maybe we can still apply your suggestion, but with a warning about adding `-t`.